CARDS-2755: [YE] New survey: outpatient

marta- opened the pull request:
WORK IN PROGRESS

Four of the five subtasks are tackled so far on this branch:
* [CARDS-2756](https://phenotips.atlassian.net/browse/CARDS-2756): _Outpatient survey: questionnaire definition_
* [CARDS-2757](https://phenotips.atlassian.net/browse/CARDS-2757): _Outpatient survey: ClinicMapping / Survey setup_
* [CARDS-2759](https://phenotips.atlassian.net/browse/CARDS-2759): _Outpatient survey email setup_
  * Email templates, sender / subject configuration, and email schedule (2 days and 5 days)
  * The templates rely on a `${date}` variable that should be the visit date, formatted as `MMMM d`, e.g. "September 9" (**TO DO**)
  * Not tested before push
* [CARDS-2760](https://phenotips.atlassian.net/browse/CARDS-2760): _Outpatient survey responses export_
  * Not tested before push 

[CARDS-2756]: https://phenotips.atlassian.net/browse/CARDS-2756?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[CARDS-2757]: https://phenotips.atlassian.net/browse/CARDS-2757?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[CARDS-2759]: https://phenotips.atlassian.net/browse/CARDS-2759?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[CARDS-2760]: https://phenotips.atlassian.net/browse/CARDS-2760?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

marta- committed:
CARDS-2755 / CARDS-2756: Outpatient survey - questionnaire definition

WIP - AI generated questionnaire definition - does not load due to oak validation errors

marta- committed:
CARDS-2755 / CARDS-2756: Outpatient survey - questionnaire definition
WIP - fixed property types, typos

marta- committed:
CARDS-2755 / CARDS-2756: Outpatient survey - questionnaire definition
WIP:
* moved hidden reference questions inside module 1
* added computed question for visit month
* removed Module 1/2/3 section labels
* added `notApplicable` property to the last answer option of questions 2a and 2b

veronikaslc committed:
CARDS-2755 / CARDS-2756: Outpatient survey - questionnaire definition
Corrected question text and options
Added defaultOrder to answer options
Fixed conditions
Added subsections
Made question 3 boolean

marta- committed:
CARDS-2755 / CARDS-2756: Outpatient survey - questionnaire definition

- Bugfix: questions ocpe_dpartment and ocpe_visit_date were referencing non-existent questions from Visit information
- Condition fixes:
  - for the Virtual care section, changed `q1 <> 0` to `q1 includes any {1,2}` to prevent the section from showing up by default
  - removed unnecessary condition `q1 <> 0` from q3's parent section
  - added `q2a is not empty` and `q2b is not empty` to q3 conditions to prevent it from showing up by default
- Made q4 a boolean question for shorter source code
- Made q16 optional

marta- committed:
CARDS-2755 / CARDS-2756: Outpatient survey - questionnaire definition
* Meaningful section ids for Virtual Care and Patient Care sections
* Moved bracketed phrase from question text to question description in q2a, q2b, q15
* Made the 2 scale questions q15 and q18 look consistent by adding ` - Neutral` to the label of value 5 in q15
* Removed bracketed words from q19 text, as they are irrelevant to patient users

marta- committed:
CARDS-2755 / CARDS-2756: Outpatient survey - questionnaire definition

Text formatting improvements

marta- committed:
CARDS-2755 / CARDS-2756: Outpatient survey - questionnaire definition
- Bugfix: q3 sometimes not displayed due to contradicting conditions
- UX improvement: only unfold the Care Experience section once type of care is selected

marta- committed:
CARDS-2755 / CARDS-2756: Outpatient survey - questionnaire definition

Split Module 1 into 3 pages

marta- committed:
CARDS-2755 / CARDS-2727: Outpatient survey - ClinicMapping / Survey setup

Hajany commented:
As noted and brought up - since Virtual Care has it's own page, for in-person visits it skips from Page 1 (Type of Visit) to Page 3 (Care Experience). I would replace the number progress to bar progress at the bottom of the pages

veronikaslc committed:
CARDS-2755: [YE] New survey: outpatient
Fixed q12, changed progress to bar

marta- committed:
CARDS-2755 / CARDS-2756: Outpatient survey - questionnaire definition

Fixed question 12.

Hajany commented:
Make this question 
<img width="1098" height="217" alt="image" src="https://github.com/user-attachments/assets/2616c8a3-4acd-4848-90f2-68e32c524d03" />

**conditional** on if "Yes" is selected in the below question
<img width="1061" height="175" alt="image" src="https://github.com/user-attachments/assets/a7072421-0f0b-4167-b902-69fc9d194546" />

Currently, we show "If a telephone or virtual visit was appropriate, would you choose a virtual visit again if you had the option to?" regardless and if No in "Were you able to continue with the telephone or virtual appointment despite the technical problems?" is selected we hide. Looks cleaner to not show until "Yes" is selected vs. hiding

Hajany commented:
Too much colour distinction in the progress bar - why are there 2 different shades of blue? I think it would look cleaner to have one shade of blue indicate progress/page user is currently on. After you click next is progresses 
<img width="1587" height="66" alt="image" src="https://github.com/user-attachments/assets/d1012f1d-dc4a-4328-aeca-5ec1ada14cc9" />

marta- commented:
> Too much colour distinction in the progress bar - why are there 2 different shades of blue? I think it would look cleaner to have one shade of blue indicate progress/page user is currently on. After you click next is progresses 

The blue in full opacity indicates the completed pages. The slightly lighter blue indicates the current page. The very light blue indicates the pages not yet reached.

This was agreed upon during our brand updates discussions and implemented in PR #2030 . Before this change, the buffer zone (current page) was depicted in secondary color (for YE that was the "uhn gold" from the previous color scheme).

marta- committed:
CARDS-2755 / CARDS-2756: Outpatient survey - questionnaire definition

Updated the conditions for displaying question 4:
- Bugfix: q4 not displayed when switching from reporting technical problems that prevented the completion of the appointment to reporting no technical problems
- Usability / flow improvement: only show q4 when all previous questions from the Virtual Care section have been answered and the q4 conditions are met

marta- commented:
> Question 4 "If a telephone or virtual visit was appropriate, would you choose a virtual visit again if you had the option to?" should still appear if they change their mind to the first question and say "there were no technical problems" - currently it does not

The conditions for displaying q4 have been updated in 2213416 .

marta- commented:
New manifestation of the same issue after the latest commit:
- Select "telephone" in q1 on page 1, click NEXT
- Select any technical problem in q2a
- Select "No" in q3 (i.e. not able to continue with the appointment)
- Click BACK
- Select "Virtual" in q1 on page 1, click NEXT

-> Note how "No" is still selected in q3 on page 2

marta- committed:
CARDS-2755 / CARDS-2759: Outpatient survey email setup

Added email templates, subject + sender configuration, and email schedule configuration.

Not tested.

marta- committed:
CARDS-2755 / CARDS-2760: Outpatient survey responses export configuration

marta- commented on modules/data-entry/src/main/frontend/src/questionnaire/Section.jsx:
This fix should have its own Jira issue and be published in a separate pull request.

marta- replied:
Created https://phenotips.atlassian.net/browse/CARDS-2766 .
